modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/TypeUtils.java

```diff
@@ -151,22 +150,12 @@ public static RelDataType combinedRowType(IgniteTypeFactory typeFactory, RelData
         return builder.build();
     }
 
-    /**
-     * CreateRowType.
-     * TODO Documentation https://issues.apache.org/jira/browse/IGNITE-15859
-     */
-    public static RelDataType createRowType(IgniteTypeFactory typeFactory, Class<?>... fields) {
-        List<RelDataType> types = Arrays.stream(fields)
-                .map(typeFactory::createJavaType)
-                .collect(Collectors.toList());
-
-        return createRowType(typeFactory, types, "$F");
+    /** Assembly output type from input types. */
+    public static RelDataType createRowType(IgniteTypeFactory typeFactory, List<RelDataType> fields) {
+        return createRowType(typeFactory, fields.toArray(new RelDataType[0]));
```

Comment:
The fact that we are converting a list to an array and then an array to a list looks a little strange. Maybe we can rework it a bit to avoid this double conversion?

Reply:
you are right, fixed.